blockchain/utxocache.go

```diff
@@ -381,52 +393,62 @@ func (c *UtxoCache) fetchEntry(dbTx database.Tx, outpoint wire.OutPoint) (*UtxoE
 
 // FetchEntry returns the specified transaction output from the utxo set.  If
 // the output exists in the cache, it is returned immediately.  Otherwise, it
-// uses an existing database transaction to fetch the output from the database,
-// cache it, and return it to the caller.  A cloned copy of the entry is
-// returned so it can safely be mutated by the caller without invalidating the
-// cache.
+// fetches the output from the backend, caches it, and returns it to the
+// caller.  A cloned copy of the entry is returned so it can safely be mutated
+// by the caller without invalidating the cache.
 //
 // When there is no entry for the provided output, nil will be returned for both
 // the entry and the error.
 //
 // This function is safe for concurrent access.
-func (c *UtxoCache) FetchEntry(dbTx database.Tx, outpoint wire.OutPoint) (*UtxoEntry, error) {
+func (c *UtxoCache) FetchEntry(outpoint wire.OutPoint) (*UtxoEntry, error) {
 	c.cacheLock.Lock()
-	entry, err := c.fetchEntry(dbTx, outpoint)
+	entry, err := c.fetchEntry(outpoint)
 	c.cacheLock.Unlock()
 	return entry, err
 }
 
 // FetchEntries adds the requested transaction outputs to the provided view.  It
 // first checks the cache for each output, and if an output does not exist in
-// the cache, it will fetch it from the database.
+// the cache, it will fetch it from the backend.
 //
 // Upon completion of this function, the view will contain an entry for each
 // requested outpoint.  Spent outputs, or those which otherwise don't exist,
 // will result in a nil entry in the view.
 //
 // This function is safe for concurrent access.
-func (c *UtxoCache) FetchEntries(filteredSet viewFilteredSet, view *UtxoViewpoint) error {
+func (c *UtxoCache) FetchEntries(filteredSet ViewFilteredSet, view *UtxoViewpoint) error {
 	c.cacheLock.Lock()
-	err := c.db.View(func(dbTx database.Tx) error {
-		for outpoint := range filteredSet {
-			entry, err := c.fetchEntry(dbTx, outpoint)
-			if err != nil {
-				return err
-			}
-
-			// NOTE: Missing entries are not considered an error here and instead
-			// will result in nil entries in the view.  This is intentionally done
-			// so other code can use the presence of an entry in the view as a way
-			// to unnecessarily avoid attempting to reload it from the database.
-			view.entries[outpoint] = entry
+	for outpoint := range filteredSet {
+		entry, err := c.fetchEntry(outpoint)
+		if err != nil {
+			return err
 		}
```

Comment:
`c.cacheLock` not unlocked on error here.

Reply:
Good catch!  Added.